lib/fluent/plugin/in_opensearch.rb

```diff
@@ -239,6 +240,43 @@ def parse_time(value, event_time, tag)
       return Time.at(event_time).to_time
     end
 
+    def get_reachable_hosts(hosts=nil)
+      reachable_hosts = []
+      attempt = 0
+      loop do
+        hosts.each do |host|
+          begin
+            if @infinite_check_connection == true
```

Comment:
```suggestion
            if @infinite_check_connection
```

`== true` is redundant because if clause already handles boolean value.